src/pushsource/_impl/source.py

```diff
@@ -29,14 +31,42 @@ class SourceUrlError(ValueError):
 
 
 class SourceWrapper(object):
+    timeout = 900
+    poll_rate = 30
     # Internal class to ensure that all source instances support enter/exit
     # for with statements even if underlying instance doesn't implement it
     # (since it originally was not mandatory).
     def __init__(self, delegate):
         self.__delegate = delegate
 
     def __iter__(self):
-        return self.__delegate.__iter__()
+        """
+        Poll until a source file is present before yielding an item.
+
+        This is necessary due to a limitation, where file changes may not immediately propagate
+        to NFS.
+
+        Raises:
+            RuntimeError:
+                If source file is still not present after time limit.
+        Yields:
+            PushItem:
+                Created push item.
+        """
+        generator = self.__delegate.__iter__()
+        for item in generator:
+            if not hasattr(item, "src") or not item.src:
+                yield item
+            else:
+                for i in range(self.timeout // self.poll_rate):
+                    if os.path.exists(item.src):
+                        break
+                    if i == (self.timeout // self.poll_rate) - 1:
+                        raise RuntimeError(
+                            "Push item source {0} is missing".format(item.src)
```

Comment:
may be a good idea to mention the timeout.

Reply:
Added